lib/Viewer.js

```diff
@@ -185,16 +187,51 @@ Viewer.prototype.importXML = function(xml, done) {
       return done(err, parseWarnings);
     }
 
+    var diagram = null;
+    if (diagramId) {
+      diagram = find(definitions.diagrams, function(element) {
+        return element.id == diagramId;
+      });
+    }
```

Comment:
This part is reoccurring in the code so it could be extracted to a separate function.